Data_Engine/Create/Table.cs

```diff
@@ -0,0 +1,152 @@
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Data;
+using System.Linq;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.Engine.Reflection;
+using BH.oM.Base;
+using BH.oM.Data.Collections;
+
+namespace BH.Engine.Data
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Generates a Table from a list of BHoMObjects. To generate a custom table, use this in conjunction with CustomObjects")]
+        [Input("name", "Name of the table")]
+        [Input("bhomObjects", "The list of BHoMObejcts to turn into a Table")]
+        [Input("ignoreName", "Toggles whether to skip the name property or not, defaults to false")]
+        [Input("ignoreGuid", "Toggles whether to skip the name Guid or not, defaults to false")]
+        [Input("ignoreTags", "Toggles whether to skip the name Tags or not, defaults to false")]
+        [Output("Table", "The created Table class")]
+        public static Table Table(string name, List<IBHoMObject> bhomObjects, bool ignoreName = false, bool ignoreGuid = true, bool ignoreTags = true)
+        {
+            IBHoMObject first = bhomObjects.First();
+
+            Dictionary<string, object> properties = first.PropertyDictionary();
+
+            List<DataColumn> columns = new List<DataColumn>();
+
+            properties.CleanUnwantedProperties(ignoreName, ignoreGuid, ignoreTags);
+
+            foreach (var kvp in properties)
+            {
+                
+                columns.Add(new DataColumn(kvp.Key, kvp.Value.GetType()));
```

Comment:
This needs a null check:

![image](https://user-images.githubusercontent.com/16853390/68446078-a04a9580-0216-11ea-9538-deb0d3ae03c6.png)